ops/src/messenger.rs

```diff
@@ -70,18 +72,35 @@ impl Messenger {
         repo: &github::Repo,
         branch: &str,
         commits: &[T],
+        thread_urls: Vec<String>,
     ) {
+        // We could look at the thread_url to see if threads have been used in the past, but that
+        // wouldn't respect the users' choice if they change the setting later.
+        let use_threads = self.config.repos().notify_use_threads(repo) && !thread_urls.clone().is_empty();
+
         for channel in self.config.repos().lookup_channels(repo, branch, commits) {
             let channel_msg = format!(
                 "{} ({})",
                 msg,
                 util::make_link(&repo.html_url, &repo.full_name)
             );
-            self.slack.send(slack::req(
-                SlackRecipient::new(&channel, &channel),
-                &channel_msg,
-                attachments,
-            ));
+            if !use_threads {
+                self.slack.send(slack::req(
+                    SlackRecipient::new(&channel, &channel),
+                    &channel_msg,
+                    attachments,
+                    None,
+                ));
+            } else {
+                for thread_url in thread_urls.clone() {
```

Comment:
also shouldn't need to clone here. 

```suggestion
                for thread_url in &thread_urls {
```